File: solutions/881.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(), people.end());
        int left = 0;
        int right = people.size() - 1;
        int count = 0;
        while (left <= right) {
            if (people[left] + people[right] > limit) {
                right--;
            }
            else {
                left++;
                right--;
            }
            count++;
        }
        return count;
    }
};

File: rust_solutions/881.rs
impl Solution {
    pub fn num_rescue_boats(people: Vec<i32>, limit: i32) -> i32 {
        let mut people = people;
        people.sort_unstable();
        let mut left: i32 = 0;
        let mut right: i32 = (people.len() - 1) as i32;

        let mut count: i32 = 0;
        while (left <= right) {
            if (people[left as usize] + people[right as usize] > limit) {
                right -= 1;
            }
            else {
                left += 1;
                right -= 1;
            }
            count += 1;
        }
        return count;
    }
}
